pkgs/development/interpreters/perl/default.nix

```diff
@@ -22,7 +22,8 @@ let
   libcInc = lib.getDev libc;
   libcLib = lib.getLib libc;
   crossCompiling = stdenv.buildPlatform != stdenv.hostPlatform;
-  common = { version, sha256 }: stdenv.mkDerivation (rec {
+
+  common = { self, version, sha256 }: stdenv.mkDerivation (rec {
```

Comment:
Instead of having to pass `self` here, you can do

```nix
{ version, sha256 }: let self = stdenv.mkDerivation (rec {
  # ...
}); in self;
```

Reply:
interesting, I might adopt this for lua interpreters too

Reply:
By the way, regarding `sha256`, we need to think about whether we want to have the specific algorithm type as parameter, now that Nix supports SRI hash format https://github.com/NixOS/nix/commit/6024dc1d97212130c19d3ff5ce6b1d102837eee6

I think for 19.09 we can add `hash` as parameter to our fetchers.

Reply:
My fetchers are directly converted by the output of the nix-prefetch tool, so if it supports SRI export, then I may use that.

Reply:
> Instead of having to pass self here, you can do

It was so initialy :)
The reasons for make self as parameter were:
1. to make the code closer to ```python```/```lua``` (to easy further refactoring 
2. to make the diff in this PR smaller and easier to review (introducing the local ```self``` would change intendation of a big code block, this would increase the number of changed lines and Github would render the diff weird)